components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1438,60 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, Unit<Length> unit) {
+    if (value == null || value.isInfinite()) {
+      LOGGER.debug("Expected float value for position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) unit = UNITS.REFERENCEFRAME;
```

Comment:
There is another benefit with these methods on top of creating `Length` objects which is the value validation e.g. floats for position or positive floats for physical sizes in agreement with the model. I also agree this is something there is some generic content we should be able to migrate upstream to the `ome-xml` component eventually. 

For now, the  goal is to reduce the logic  implemented in every reader in favor of a more generic :

```
Length posx = FormatTools.getPosition(DataTools.parseDouble(value), UNITS.REFERENCE_FRAME);
```